dpar/src/models/tensorflow/collector.rs

```diff
@@ -95,10 +114,22 @@ where
 {
     fn collect(&mut self, t: &T::Transition, state: &ParserState) -> Result<(), Error> {
         // Lazily add a new batch tensor.
+
+        let n_deprels = self
+            .vectorizer
+            .layer_lookups()
+            .layer_lookup(Layer::DepRel)
+            .unwrap()
+            .len();
+
         if self.instance_idx == 0 {
             let layer_tensors = self.new_layer_tensors(self.batch_size);
-            self.inputs.push(layer_tensors);
+            self.lookup_inputs.push(layer_tensors);
             self.labels.push(Tensor::new(&[self.batch_size as u64]));
+            self.non_lookup_inputs
+                .push(Tensor::new(&[
```

Comment:
Shouldn't this be:

`&[self.batch_size, n_deprels * T::ATTACHMENT_ADDRS.len()]`

?